src/components/Course/Course.vue

```diff
@@ -81,6 +81,8 @@ export default defineComponent({
     active: { type: Boolean, required: true },
     isReqCourse: { type: Boolean, required: true },
     semesterIndex: { type: Number, required: false, default: 0 },
+    season: { type: String, required: false, default: '' },
+    year: { type: Number, required: false, default: 0 },
```

Comment:
I think making these properties optional to fix the type error is kind of sketchy. Is there a reason why they might not be needed sometimes?

Reply:
Because in the implementation of retrieving a course, if a valid season and year is not provided for the course, we will retrieve the latest offering of that course. So by having a default value, we are pretty much saying that we want the latest offering of the course.